var/spack/repos/builtin/packages/sgpp/package.py

```diff
@@ -0,0 +1,167 @@
+    version('3.2.0', sha256='dab83587fd447f92ed8546eacaac6b8cbe65b8db5e860218c0fa2e42f776962d')
+    # Versions with Python 2 bindings:
+    version('3.1.0', sha256='6b46bc5b3966e92567d6754130666bdffb7be1d1d2c1b427d7ce964b8eaab526')
+    version('3.0.0', sha256='4dd9049e664abd7db78c355fea5e192167812f443115d4bf686a51bb1e9bda9c')
+
+    # Patch that ensures libraries will actually
+    # be copied into prefix/lib upon installation
+    # (otherwise it would be prefix/sgpp/lib)
+    patch('directory.patch', when='@:3.2.0')
+    # Fix faulty setup.py in 3.2.0
+    patch('fix-setup-py.patch', when='@3.2.0')
+    # Backport opencl fix
+    patch('ocl.patch', when='@:3.2.0+opencl')
+
+    variant('simd',
+            default='sse3',
+            values=('sse3', 'sse42', 'avx', 'fma4', 'avx2', 'avx512'),
+            description='Specifies the SIMD extension to be used')
+    variant('python', default=True,
+            description='Provide Python bindings for SGpp')
+    variant('java', default=False,
+            description='Provide Java bindings for SGpp')
+    # export CLASSPATH=$(spack location --install-dir sgpp)/lib/jsgpp.jar
+    # After that one can compile and run simple java examples with sgpp
+    variant('optimization', default=True,
+            description='Builds the optimization module of SGpp')
+    variant('pde', default=True,
+            description='Builds the datadriven module of SGpp')
+    variant('quadrature', default=True,
+            description='Builds the datadriven module of SGpp')
+    variant('datadriven', default=False,
+            description='Builds the datadriven module of SGpp')
+    variant('misc', default=False,
+            description='Builds the misc module of SGpp')
+    variant('combigrid', default=False,
+            description='Builds the combigrid module of SGpp')
+    variant('solver', default=True,
+            description='Builds the solver module of SGpp')
+    variant('opencl', default=False,
+            description='Enables support for OpenCL accelerated operations')
+    variant('mpi', default=False,
+            description='Enables support for MPI-distributed operations')
+
+    # Mandatory dependencies
+    depends_on('scons@2.5.1', when='@:3.1.0', type=('build'))
+    depends_on('scons@3:', when='@3.2.0:', type=('build'))
+    depends_on('zlib', type=('build', 'run'))
+    # Python dependencies
+    extends('python', when='+python')
+    # Python 3 support was added in version 3.2.0
+    depends_on('python@2.7:2.8', when='@:3.1.0+python', type=('build', 'run'))
+    depends_on('python@3:', when='@3.2.0:+python', type=('build', 'run'))
+    depends_on('swig@3:', when='+python', type=('build'))
+    # Java dependencies
+    depends_on('swig@3:', when='+java', type=('build'))
+    depends_on('openjdk', when='+java', type=('build', 'run'))
+    # Python libraries (version depends on whether we use Python 2 or 3)
```

Comment:
You're missing a build dependency on `py-setuptools`

Reply:
Fixed in dcfd800